src/vm/jithelpers.cpp

```diff
@@ -2395,7 +2395,7 @@ TypeHandle::CastResult STDCALL ObjIsInstanceOfNoGC(Object *pObject, TypeHandle t
     return pMT->CanCastToClassOrInterfaceNoGC(toTypeHnd.AsMethodTable());
 }
 
-BOOL ObjIsInstanceOf(Object *pObject, TypeHandle toTypeHnd)
+BOOL ObjIsInstanceOf(Object *pObject, TypeHandle toTypeHnd, BOOL canThrowCustomInvalidCast)
```

Comment:
It does not feel like a good design to let the method take care of some of the invalid cast exceptions. I think it would be better to take care of throwing of all invalid cast exceptions when this flag is set.